src/asmcnc/apps/maintenance_app/widget_maintenance_brush_use.py

```diff
@@ -149,7 +150,31 @@ def __init__(self, **kwargs):
         self.update_strings()
         
     def restore(self):
```

Comment:
I think these functions could be refactored and broken up a bit into: 
- a function (or sequence of functions) that read the info from the spindle (and this could probs go into router_machine)
- UX side
That would make it a little bit more readable, and mean that the read in function was available in router_machine if it was needed elsewhere (e.g. as part of the prompt to replace brushes)

Reply:
Had a go at refactoring this but it doesn't look much more readable to me, does it look right to you?

Reply:
Nevermind, after testing it seems that the refactoring just breaks the functionality and I can't work out why, I think it would be best to just leave it as is. Probably best not to split wait popups up from the functionality that they are waiting for, unless there's a smarter method than what I was doing

Reply:
Ok no worries, well done for having a stab at it anyway :) 

There probably is a solution, but if it's not easy to find let's leave it for now.